### PR TITLE
build: update to latest build infrastructure

### LINKED ISSUE #236: Update micronaut-graphql build to new infrastructure
### Issue description

Update build to new infrastructure

### PR DESCRIPTION
**Pull request by timyates**
Fixes #236

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -20,3 +22,6 @@ developers=Marcel Overdijk
 
 githubCoreBranch=3.3.x
 bomProperty=micronautGraphQLVersion
+
+org.gradle.caching=true
+org.gradle.parallel=true
```
[the remark this answers is not in the document]
Comment (melix) [reply gradle.properties:27]: Parallel execution is disabled in the template, for cases where heavy integration tests could cause load issues (for example with test containers executed concurrently in different projects). But in general, if you follow the new build pattern, it should be safe to enable both caching and parallel.

**Comment by timyates** [comment]
Investigating the graal failures

**Comment by timyates** [comment]
@melix Thank you so much... 😄 (again!) 😎

**Review by melix — APPROVED** [review]

File: graphql/build.gradle [hunk 2]
```diff
@@ -1,17 +1,21 @@
+plugins {
+    id "io.micronaut.build.internal.module"
+}
+
 dependencies {
-    annotationProcessor("io.micronaut:micronaut-inject-java:${micronautVersion}")
-    annotationProcessor("io.micronaut:micronaut-validation:${micronautVersion}")
+    annotationProcessor("io.micronaut:micronaut-inject-java")
```
Comment (sdelamo) [inline graphql/build.gradle:6]: I think `annotationProcessor("io.micronaut:micronaut-inject-java")` is added automatically by the `io.micronaut.build.internal.module` plugin

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: graphql/build.gradle (hunk 2 above)
Comment (melix) [reply graphql/build.gradle:6]: That's correct: https://github.com/micronaut-projects/micronaut-build/blob/278c63890a5457637a66eaef8d6dfe59addcf291/src/main/groovy/io/micronaut/build/MicronautModulePlugin.groovy#L22-L32

Comment (timyates) [reply graphql/build.gradle]: 👍 

I'm going to try to burn that into my memory 😀

**Review by sdelamo — APPROVED** [review]

**Pull request merged**